### PR TITLE
feat: created logs saving logic and added to timer

### PR DESCRIPTION
**Pull request by wawrzonnn**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
The task was to save a log for each completed activity (Focus or Break) with all related information. The log should be persisted in local storage.

### Requirements
1. After completing a started activity (Focus or Break), the app should save a log with all related information.
2. Local storage must be used for data persistence.

### Required log schema
```javascript
interface Log {
  mode: 'BREAK' | 'FOCUS'; // Current mode that was logged
  time: number; // Time value in seconds
  createdAt: Date; // Current day timestamp
}
```

### Code Changes Review

#### src/components/Timer.tsx
The changes in this file seem to be related to the UI and not directly related to the task requirements. The changes are minor and involve adjusting the position of an element and adding a space in a formatted string.

#### src/store/store.ts
The changes in this file are directly related to the task requirements. When the timer for either Focus or Break reaches zero, a new log object is created with the required schema and saved. The `isFocusCompleted` and `isBreakCompleted` flags are used to ensure that the log is only saved once per completed activity.

However, there seems to be a missing piece in the provided code changes. The `saveLog` function is called to save the new log, but its implementation is not provided in the changes. This function should handle saving the log to local storage, which is a requirement of the task.

Without seeing the implementation of `saveLog`, it's impossible to confirm whether the code changes fully cover the task requirements. The creation of the log object seems to be correct, but we need to ensure that these logs are being correctly saved to local storage.

### Suggestions
Please provide the implementation of `saveLog` function for a complete review. This function should take a log object as a parameter and save it to local storage.

File: src/store/store.ts [hunk 1]
```diff
@@ -30,10 +30,19 @@ export const useStore = create<StoreState>((set, get) => ({
...
-			const { focusTime, breakTime, mode, initialBreakTime, initialFocusTime, startOvertime, isOvertimeRunning } = get()
+			const { focusTime, breakTime, mode, initialBreakTime, initialFocusTime, isBreakCompleted, isFocusCompleted, saveLog } = get()
 			if (mode === 'focus') {
 				// FOCUS LOGIC
 				if (focusTime.minutes === 0 && focusTime.seconds === 0) {
+					if (!isFocusCompleted) {
+						//CREATING AND SAVING COMPLETED FOCUS LOG
+						const newLog: Log = {
+							mode: "FOCUS",
+							time: initialFocusTime.minutes * 60 + initialFocusTime.seconds,
```
Comment (Sumick) [inline src/store/store.ts:50]: Where Prettier?!

Comment (wawrzonnn) [reply src/store/store.ts:50]: aronoł

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
We need to be able to save log for each completed activity

### Requirements
1. After completing started activity (Focus or Break) app should save a log with all related information.
2. Local storage must be used for data persistence.

### Required log schema
```javascript
interface Log {
  mode: 'BREAK' | 'FOCUS'; // Current mode that was logged
  time: number; // Time value in seconds
  createdAt: Date; // Current day timestamp
}
```

### Code Changes Review

#### src/components/Timer.tsx
The changes in this file are not related to the task requirements. They are only style changes.

#### src/store/store.ts
The changes in this file cover the task requirements. The `saveLog` function is implemented and it saves logs to the local storage. The logs are saved when the focus time or break time reaches zero, which means the activity is completed. The log contains the mode, time, and createdAt fields as required.

#### src/store/types.ts
The changes in this file do not match the task requirements. The mode in the Log interface should be 'BREAK' or 'FOCUS' according to the requirements, but it was changed to 'focus' or 'break'. This needs to be corrected.

### Suggestions
The `saveLog` function could be improved by checking if the log is successfully saved to the local storage. Also, the console.log statement could be removed after testing is completed.

```javascript
saveLog: (log: Log) => {
  const logs = JSON.parse(localStorage.getItem('logs') || '[]')
  logs.push(log)
  localStorage.setItem('logs', JSON.stringify(logs))
  const savedLogs = JSON.parse(localStorage.getItem('logs') || '[]')
  if (savedLogs.includes(log)) {
    console.log('Log saved successfully:', log)
  } else {
    console.error('Failed to save log:', log)
  }
}
```

### Conclusion
The code changes mostly cover the task requirements. The only issue is the mode in the Log interface in src/store/types.ts file. After correcting this, the code changes should fully cover the task requirements.

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task Goal
The task goal is to save a log for each completed activity (Focus or Break). The log should be saved in local storage and should follow the provided log schema.

### Code Changes
The code changes are made in three files: `src/components/Timer.tsx`, `src/store/store.ts`, and `src/store/types.ts`.

#### src/components/Timer.tsx
The changes in this file are minor and do not directly relate to the task requirements. They involve changing the position of a styled component and adding a space character to a formatted string. These changes do not affect the functionality of saving logs.

#### src/store/types.ts
The changes in this file involve replacing the string union type `'BREAK' | 'FOCUS'` with an enum `PomodoroMode`. This is a good change as it provides better type safety and code readability.

#### src/store/store.ts
The main logic for saving logs is added in this file. The `saveLog` function is implemented which saves a log to local storage. The function is called when the timer for Focus or Break reaches zero, creating a new log and saving it.

### Review against Task Requirements
1. After completing started activity (Focus or Break) app should save a log with all related information: The code changes cover this requirement. When the timer for Focus or Break reaches zero, a new log is created with all related information and saved to local storage.
2. Local storage must be used for data persistence: The code changes cover this requirement. The `saveLog` function saves logs to local storage.

### Suggestions
The code changes cover the task requirements. However, there is a console.log statement in the `saveLog` function which might have been used for debugging and should be removed for production code.

```diff
-saveLog: (log: Log) => {
+saveLog: (log: Log) => {
	const logs = JSON.parse(localStorage.getItem('logs') || '[]')
	logs.push(log)
	localStorage.setItem('logs', JSON.stringify(logs))
-	console.log('saved log:', log)
}
```

### Conclusion
The code changes cover the task requirements. After removing the console.log statement, the code is ready for production.

**Pull request merged**